shapeshift/web
# feat: add FOX LP APR hooks

## Description

This adds a hook (with related utility hooks) to fetch the APR from ETH/FOX UniV2.
This effectively also brings the base smart contract-interacting hooks that will be able to be used as groundwork for farming opportunities.

Most of these are adapted implementations from foxfarm, with:

- added missing typings
- added loading state
- improved naming
- logic abstracted, and simplified to remove checks we do not need, since we are in a fully controlled domain of plugins and these hooks are not intended to be consumed from anywhere else
- use of 0. percentage for consistency with the way we represent percentages in web, vs. foxfarm using the human-readable percentage directly

Note that this isn't wired up anywhere yet, but see the attached screenshot for a runtime console log of the APR data. 

## Notice




## Pull Request Type


## Issue (if applicable)

Partially tackles #1834

## Risk

None since this isn't added as app code yet. 

## Testing

- Consume `useLpApr` from e.g `src/plugins/foxPage/components/Layout.tsx`
- Verify that loading state is working
- Verify that the LP APR is matching the one in https://fox.shapeshift.com/fox-farming

## Screenshots (if applicable)

<img width="373" alt="image" src="https://user-images.githubusercontent.com/17035424/171203674-d34e97df-6012-4a56-b0e4-9dcb4dfda917.png">
